cockroachdb/datadog_checks/cockroachdb/cockroachdb.py

```diff
@@ -0,0 +1,110 @@
+                    'namespace': 'cockroachdb',
+                    'metrics': [METRIC_MAP],
+                    'send_histograms_buckets': True,
+                }
+            },
+            default_namespace='cockroachdb',
+        )
+
+    def check(self, instance):
+        scraper_config = self.get_scraper_config(instance)
+
+        if 'prometheus_url' not in scraper_config:
+            raise CheckException(
+                'You have to define at least one `prometheus_url`.'
+            )
+
+        if not scraper_config.get('metrics_mapper'):
+            raise CheckException(
+                'You have to collect at least one metric from the endpoint `{}`.'.format(
+                    scraper_config['prometheus_url']
+                )
+            )
+
+        tracked_metrics = scraper_config.get('_tracked_metrics')
+        if tracked_metrics is None:
+            tracked_metrics = scraper_config['_tracked_metrics'] = defaultdict(dict)
+        else:
+            tracked_metrics.clear()
+
+        self.process(
+            scraper_config,
+            metric_transformers={
+                metric: self.track_metric
+                for metric in TRACKED_METRICS
+            }
+        )
+
+        tags = instance.get('tags', [])
+        self.check_disk_space(tracked_metrics, instance, tags)
+        self.check_sql_execute(tracked_metrics, tags)
+
+    def check_disk_space(self, tracked_metrics, instance, tags):
+        capacity_total = tracked_metrics.get('capacity')
+        capacity_available = tracked_metrics.get('capacity_available')
+
+        if capacity_total is None or capacity_available is None:
+            self.log.info(
+                'Missing `capacity` and/or `capacity_available` metrics, skipping disk space check...'
+            )
+            return
+
+        percent_remaining = capacity_available / capacity_total * 100
+
+        disk_space_critical = int(instance.get('disk_space_critical', 5))
+        if percent_remaining <= disk_space_critical:
+            self.service_check(self.SERVICE_CHECK_DISK_SPACE, self.CRITICAL, tags=tags)
+            return
+
+        disk_space_warning = int(instance.get('disk_space_warning', 15))
+        if percent_remaining <= disk_space_warning:
+            self.service_check(self.SERVICE_CHECK_DISK_SPACE, self.WARNING, tags=tags)
+            return
+
+        self.service_check(self.SERVICE_CHECK_DISK_SPACE, self.OK, tags=tags)
+
+    def check_sql_execute(self, tracked_metrics, tags):
+        active_connections = tracked_metrics.get('sql_conns')
+        active_queries = tracked_metrics.get('sql_query_count')
+
+        if active_connections is None or active_queries is None:
+            self.log.info(
+                'Missing `sql_conns` and/or `sql_query_count` metrics, skipping sql execution check...'
+            )
+            return
+
+        if active_connections > 0 and active_queries == 0:
```

Comment:
why is this critical? This seems like it could flap just in the course of a normal application, sometimes things might just not be connected to the database, right?

Reply:
https://www.cockroachlabs.com/docs/stable/monitoring-and-alerting.html#node-is-not-executing-sql

Reply:
But should we do this in the check or let customers create monitors?

Reply:
Basically I don't think we should be doing things we can do with monitors with service checks.